src/utils.jl

```diff
@@ -25,7 +25,7 @@ end
 dot_matwise(x::AbstractMatrix, y::AbstractArray) = dot_matwise(y, x)
 
 function dot_vecwise(x::AbstractMatrix, y::AbstractMatrix)
-    return [dot(u, v) for (u, v) in zip(eachcol(x), eachcol(y))]
+    Compat.@compat return [dot(u, v) for (u, v) in zip(eachcol(x), eachcol(y))]
```

Comment:
As mentioned, I think it would be better to not use Compat at all but rewrite the code. In any case, you don't have to use `@compat` - the macro is only useful if some Julia syntax has changed and has to be rewritten before it is compiled. Here it is sufficient to load Compat.

Reply:
Yep, rewriting this too. My initial thoughts were that the `eachcol` notation was quite compact, but now that you mention it I agree it's not worth an additional dependency